optimization/ParetoOptimization/matRad_paretoOptimization.m

```diff
@@ -0,0 +1,577 @@
+if isequal(scen4D,'all')
+    scen4D = 1:size(dij.physicalDose,1);
+end
+
+linIxDIJ = find(~cellfun(@isempty,dij.physicalDose(scen4D,:,:)))';
+
+FLAG_CALC_PROB = false;
+FLAG_ROB_OPT   = false;
+
+
+for i = 1:size(cst,1)
+    for j = 1:numel(cst{i,6})
+       if strcmp(cst{i,6}{j}.robustness,'PROB') && numel(linIxDIJ) > 1
+          FLAG_CALC_PROB = true;
+       end
+       if ~strcmp(cst{i,6}{j}.robustness,'none') && numel(linIxDIJ) > 1
+          FLAG_ROB_OPT = true;
+       end
+    end
+end
+
+if FLAG_CALC_PROB
+    [dij] = matRad_calculateProbabilisticQuantities(dij,cst,pln);
+end
+
+
+% set optimization options
+if ~FLAG_ROB_OPT || FLAG_CALC_PROB     % if multiple robust objectives are defined for one structure then remove FLAG_CALC_PROB from the if clause
+   ixForOpt = scen4D;
+else
+   ixForOpt = linIxDIJ;
+end
+
+switch pln.bioParam.quantityOpt
+    case 'effect'
+        backProjection = matRad_EffectProjection;
+    case 'RBExD'
+        %Capture special case of constant RBE
+        if strcmp(pln.bioParam.model,'constRBE')
+            backProjection = matRad_ConstantRBEProjection;
+        else
+            backProjection = matRad_VariableRBEProjection;
+        end
+    case 'physicalDose'
+        backProjection = matRad_DoseProjection;
+    otherwise
+        warning(['Did not recognize bioloigcal setting ''' pln.probOpt.bioOptimization '''!\nUsing physical dose optimization!']);
+        backProjection = matRad_DoseProjection;
+end
+
+%Give scenarios used for optimization
+backProjection.scenarios    = ixForOpt;
+backProjection.scenarioProb = pln.multScen.scenProb;
+
+optiProb = matRad_OptimizationProblem(backProjection,cst);
+optiProb.quantityOpt = pln.bioParam.quantityOpt;
+if isfield(pln,'propOpt') && isfield(pln.propOpt,'useLogSumExpForRobOpt')
+    optiProb.useLogSumExpForRobOpt = pln.propOpt.useLogSumExpForRobOpt;
+end
+
+%Get Bounds
+if ~isfield(pln.propOpt,'boundMU')
+    pln.propOpt.boundMU = false;
+end 
+
+if pln.propOpt.boundMU
+    if (isfield(dij,'minMU') || isfield(dij,'maxMU')) && ~isfield(dij,'numParticlesPerMU')
+        matRad_cfg.dispWarning('Requested MU bounds but number of particles per MU not set! Bounds will not be enforced and standard [0,Inf] will be used instead!');
+    elseif ~isfield(dij,'minMU') && ~isfield(dij,'maxMU')
+        matRad_cfg.dispWarning('Requested MU bounds but machine bounds not defined in dij.minMU & dij.maxMU! Bounds will not be enforced and standard [0,Inf] will be used instead!');
+    else
+        if isfield(dij,'minMU')
+            optiProb.minimumW = dij.numParticlesPerMU .* dij.minMU / 1e6;
+            matRad_cfg.dispInfo('Using lower MU bounds provided in dij!\n')
+        end
+
+        if isfield(dij,'maxMU')
+            optiProb.maximumW = dij.numParticlesPerMU .* dij.maxMU / 1e6;
+            matRad_cfg.dispInfo('Using upper MU bounds provided in dij!\n')
+        end
+    end
+else
+    matRad_cfg.dispInfo('Using standard MU bounds of [0,Inf]!\n')
+end
```

Comment:
you put a lot of effort in externalization optimization initialization but you here do everything within the function.